packages/cactus-plugin-ledger-connector-iroha/src/main/typescript/transaction.ts

```diff
@@ -0,0 +1,491 @@
+import { Logger } from "@hyperledger/cactus-common";
+import {
+  LoggerProvider,
+  LogLevelDesc,
+  Http405NotAllowedError,
+} from "@hyperledger/cactus-common";
+import {
+  IrohaCommand,
+  IrohaQuery,
+  RunTransactionRequestV1,
+  RunTransactionResponse,
+} from "./generated/openapi/typescript-axios";
+
+import { RuntimeError } from "run-time-error";
+import * as grpc from "grpc";
+
+import {
+  GrantablePermission,
+  GrantablePermissionMap,
+} from "iroha-helpers-ts/lib/proto/primitive_pb";
+
+import { CommandService_v1Client as CommandService } from "iroha-helpers-ts/lib/proto/endpoint_grpc_pb";
+import { QueryService_v1Client as QueryService } from "iroha-helpers-ts/lib/proto/endpoint_grpc_pb";
+
+import commands from "iroha-helpers-ts/lib/commands/index";
+import queries from "iroha-helpers-ts/lib/queries";
+
+export class Transaction {
+  private readonly log: Logger;
+  public static readonly CLASS_NAME = "Transaction";
+
+  public get className(): string {
+    return Transaction.CLASS_NAME;
+  }
+
+  constructor(logLevel?: LogLevelDesc) {
```

Comment:
@StepniowskiP Please use an interface to define an object that contains the options, that way adding new arguments can be done within that object instead of having to pile up new options as the Nth constructor parameter and struggle with ordering, etc.

Reply:
Done